app/assets/javascripts/discourse/app/lib/color-scheme-picker.js

```diff
@@ -49,32 +50,51 @@ export function listColorSchemes(site, options = {}) {
 export function loadColorSchemeStylesheet(
   colorSchemeId,
   theme_id,
-  dark = false
+  darkMode = false
 ) {
   const themeId = theme_id ? `/${theme_id}` : "";
   ajax(`/color-scheme-stylesheet/${colorSchemeId}${themeId}.json`).then(
     (result) => {
       if (result && result.new_href) {
-        const elementId = dark ? "cs-preview-dark" : "cs-preview-light";
+        const elementId = darkMode ? "cs-preview-dark" : "cs-preview-light";
         const existingElement = document.querySelector(`link#${elementId}`);
         if (existingElement) {
           existingElement.href = result.new_href;
         } else {
           let link = document.createElement("link");
           link.href = result.new_href;
-          link.media = dark
+          link.media = darkMode
             ? "(prefers-color-scheme: dark)"
             : "(prefers-color-scheme: light)";
           link.rel = "stylesheet";
           link.id = elementId;
 
           document.body.appendChild(link);
         }
+        if (!darkMode) {
+          later(() => {
+            const schemeType = getComputedStyle(document.body).getPropertyValue(
+              "--scheme-type"
+            );
+
+            if (schemeType.trim() === "dark") {
+              setDarkLogoMediaAttribute("");
+            } else {
+              setDarkLogoMediaAttribute("(prefers-color-scheme: dark)");
+            }
+          }, 500);
+        }
       }
     }
   );
 }
 
+export function setDarkLogoMediaAttribute(value = "") {
+  document
+    .querySelector(`picture source.dark-logo`)
+    ?.setAttribute("media", value);
```

Comment:
I think this element is originally rendered by the `home-logo` widget? Is modifying the DOM manually going to cause any weird side effects? e.g. if the header is re-rendered for some other reason, will this change be overridden?

I wonder if instead we should modify some central state  (`session.defaultColorSchemeIsDark`, maybe?), and then make the header [re-render whenever that property changes](https://github.com/discourse/discourse/blob/f57d49ed01c0b3fad6cb68408297f6c22f33e1e0/app/assets/javascripts/discourse/app/components/site-header.js#L29-L36)?

Reply:
Great point, I have updated the PR using an update to the central state. The second session state, `darkModeAvailable`, is needed when resetting the dark color scheme (i.e. choosing "Same as regular").